ibis/sql/compiler.py

```diff
@@ -1330,19 +1336,16 @@ def __init__(self, table_set, select_set,
         self.indent = indent
 
         self.result_handler = result_handler
-        self.params = params
 
-    translator = None
-
-    def _translate(self, expr, context=None, named=False,
-                   permit_subquery=False):
-        if context is None:
-            context = self.context
+    @property
+    def translator(self):
+        return self.context.dialect.translator
```

Comment:
It's quite hard to distinguish between Context, Dialect and Translator. Too much indirection for me.
What are the relationship between them?

Reply:
I'll add some documentation for this in `ibis/sql/compiler.py`.

Reply:
Thanks :)